backend/app/controllers/uoas.js

```diff
@@ -162,139 +168,36 @@ module.exports = {
-                                    'AND "productId" = ' + req.body.productId
-                                );
-
-                                // Soft delete the task with that UAO ID
-                                yield thunkQuery(
-                                    'UPDATE "Tasks"' +
-                                    'SET "isDeleted" = (to_timestamp(' + Date.now() +
-                                    '/ 1000.0)) WHERE "productId" = ' + req.body.productId +
-                                    'AND "uoaId" = ' + req.params.id
-                                );
-
-                                yield thunkQuery(
-                                    'UPDATE "UnitOfAnalysis"' +
-                                    'SET "isDeleted" = (to_timestamp(' + Date.now() +
-                                    '/ 1000.0)) WHERE "id" = ' + req.params.id
-                                );
-
-                                yield common.bumpProjectLastUpdated(req, project[0].id);
-                            }
-                        }
-                    }
-                } else {
-                    throw new HttpError(403, 'No Project Found');
-                }
-            } else {
-                // TODO: Replicate checks for conditions if there is no productID passed in.
-                // Soft delete the UOA from the Product UAO Table
-                yield thunkQuery(
-                    'UPDATE "ProductUOA"' +
-                    'SET "isDeleted" = (to_timestamp(' + Date.now() +
-                    '/ 1000.0)) WHERE "UOAid" = ' + req.params.id
-                );
-
-                // Soft delete the task with that UAO ID
-                yield thunkQuery(
-                    'UPDATE "Tasks"' +
-                    'SET "isDeleted" = (to_timestamp(' + Date.now() +
-                    '/ 1000.0)) WHERE "uoaId" = ' + req.params.id
-                );
+            //Check if the UOA passed in has a productId associated with it.
+            const productUOA = yield thunkQuery(
+                ProductUOA.select().from(ProductUOA).where(ProductUOA.UOAid.equals(req.params.id))
+            );
 
+            if (!_.first(productUOA)) {
                 // Soft delete from the UOA table
                 yield thunkQuery(
                     'UPDATE "UnitOfAnalysis"' +
                     'SET "isDeleted" = (to_timestamp(' + Date.now() +
                     '/ 1000.0)) WHERE "id" = ' + req.params.id
                 );
+            } else {
+                if (req.body.productId) {
+                    const productId = [{productId: req.body.productId}];
+                    yield * uoaSoftDeleteHelper(req, productId, DELETE_OPTIONS.projectOnly);
+                } else {
+                    if (productUOA.length === 1) { // UOA is assigned to only one project
+                        yield * uoaSoftDeleteHelper(req, productUOA, DELETE_OPTIONS.entireSystem);
```

Comment:
This line and line 197 are identical. Should these two cases be doing different things?

Reply:
hmm weird, I was doing something slightly different before it seems. Corrected it!